edb/edgeql/compiler/stmt.py

```diff
@@ -1022,8 +1023,28 @@ def compile_result_clause(
         if result_alias:
             # `SELECT foo := expr` is equivalent to
             # `WITH foo := expr SELECT foo`
+            rexpr = astutils.ensure_ql_select(result_expr)
+            if (
+                sctx.implicit_limit
+                and rexpr.limit is None
+                and not sctx.inhibit_implicit_limit
+            ):
+                # Inline alias is special: it's both "exposed",
+                # but also subject for further processing, so
+                # make sure we don't mangle it with an implicit
+                # limit.
```

Comment:
Hm, so does this not get limited at all, then?

Reply:
It does, by the outer query.

```
SELECT 
   _ := (foo)  # <- implicit nested SELECT here, don't limit
LIMIT
   <implicit>
```